### ISSUE TITLE
Namespaces are not respected

### ISSUE THREAD

**Issue by tamird** [alexeyxo/protobuf-swift#24]
Hello! Thanks for maintaining this project. We're evaluating using this for our iOS apps and have run into a filename-collision issue, which seems to be caused by `protobuf-swift` not respecting package namespaces. Repro:

foo/person.proto:
```protobuf
package foo;

message Person {
 optional uint32 age = 1;
}
```
bar/person.proto:
```protobuf
package bar;

message Person {
 optional uint32 age = 1;
}
```
output:
```
~ protoc --plugin=../../protobuf-swift/src/compiler/protoc-gen-swift {foo,bar}/person.proto --swift_out=../../swift_protos
Person.pb.swift: Tried to write the same file twice.
~ echo $?
1
```

**Comment by alexeyxo** [comment]
I am agree, but i don't know how do it task. In the swift and objective-c syntax has not "namespaces".  I would think... [Issue #27](https://github.com/alexeyxo/protobuf-objc/issues/27)

**Comment by tamird** [comment]
https://twitter.com/clattner_llvm/status/474730716941385729

Seems like you need a separate xcode target per package.

**Comment by mhahn** [comment]
+1. I had to restructure my protobufs so all the messages were nested to avoid collisions. respecting the package would be huge.

**Comment by alexeyxo** [comment]
You can test. 
Branch name "package".

**Comment by tamird** [comment]
Can you open a pull request?

**Comment by alexeyxo** [comment]
I have not finished. I should to clean up the code.

**Comment by alexeyxo** [comment]
Fixed f612ae31a4726d10f6f863e1f0e0195f03bcc340
